source/demo.tsx

```diff
@@ -1,18 +1,22 @@
 /* eslint-disable no-console */
 import React from 'react'
-import ReactDOM from 'react-dom'
+import { createRoot } from 'react-dom/client'
```

Comment:
This is the only substantial change here, to support the latest rendering method. Could we please refer the rest? The demo script is designed to go with it's host html - I don't see much reason to safety check whether that hard coded page has an element on it. Also, this change appears to have altered the indentation and formatting quite a bit